src/testflinger_device_connectors/devices/oemrecovery/oemrecovery.py

```diff
@@ -114,18 +116,90 @@ def copy_ssh_id(self):
+        try:
+            test_username = self.job_data.get("test_data", {}).get(
+                "test_username", "ubuntu"
+            )
+            test_password = self.job_data.get("test_data", {}).get(
+                "test_password", "ubuntu"
+            )
+        except AttributeError:
+            test_username = "ubuntu"
+            test_password = "ubuntu"
+
+        while True:
+            try:
+                image_type = (
+                    subprocess.check_output(
+                        (
+                            "sshpass -p {} ssh -o StrictHostKeyChecking=no "
+                            "-o UserKnownHostsFile=/dev/null "
+                            "{}@{} snap model | grep model"
+                        ).format(
+                            test_password,
+                            test_username,
+                            self.config["device_ip"],
+                        ),
+                        shell=True,
+                    )
+                ).decode()
+                break
+
+            except subprocess.SubprocessError:
+                time.sleep(10)
+                continue
+
+        if "generic-classic" in image_type:
+            while True:
+                # We are currently not using this method on
+                # classic image, so not doing much here.
+                try:
+                    self.copy_ssh_id()
+                    return
+                except subprocess.SubprocessError:
+                    time.sleep(10)
+                    continue
+
+        else:
+            while True:
+                try:
+                    boot_state = (
+                        subprocess.check_output(
+                            (
+                                "sshpass -p {} ssh "
+                                "-o StrictHostKeyChecking=no "
+                                "-o UserKnownHostsFile=/dev/null "
+                                "{}@{} snap changes "
+                                '| grep "Initialize device"'
+                            ).format(
+                                test_password,
+                                test_username,
+                                self.config["device_ip"],
+                            ),
+                            shell=True,
+                        )
+                    ).decode()
+
+                    if "Done" in boot_state:
+                        self.copy_ssh_id()
+                        return
+                except subprocess.SubprocessError:
+                    time.sleep(10)
+
     def check_device_booted(self):
         """Check to see if the device is booted and reachable with ssh"""
         logger.info("Checking to see if the device is available.")
-        started = time.time()
+
         # Wait for provisioning to complete - can take a very long time
-        while time.time() - started < 3600:
-            try:
-                time.sleep(90)
-                self.copy_ssh_id()
```

Comment:
So here's what I'm thinking, which I don't think I was very clear about before.
1. Keep this section and eliminate wrapt_timeout_decorator
2. convert the two checks above in check_boot_state() into two separate... for instance, you might make them check_generic_classic() and check_device_initialized(). The *only* thing these need to do is run their check commands over ssh.
3. after the self.copy_ssh_id() we know that we can ssh and that the keys are there - so all we need to do at that point is do:
```
if self.check_generic_classic() or self.check_device_initialized():
    return True
```
I think this will be much simpler, what do you think?

Reply:
I think this is definitely better, I will make the change.